crankshaft-engine/src/task/resources.rs

```diff
@@ -134,23 +173,31 @@ impl From<&Defaults> for Resources {
     fn from(defaults: &Defaults) -> Self {
         Self {
             cpu: defaults.cpu(),
-            preemptible: Default::default(),
+            cpu_limit: defaults.cpu(),
             ram: defaults.ram(),
+            ram_limit: defaults.ram_limit(),
             disk: defaults.disk(),
+            preemptible: Default::default(),
             zones: Default::default(),
         }
     }
 }
 
 impl From<&Resources> for HostConfig {
     fn from(resources: &Resources) -> Self {
-        let mut host_config = HostConfig::default();
+        let mut host_config = Self::default();
+
+        // Note: Docker doesn't have a CPU reservation for containers
+        if let Some(cpu) = resources.cpu_limit() {
+            host_config.nano_cpus = Some((cpu * 1_000_000_000.0) as i64);
+        }
+
         if let Some(ram) = resources.ram() {
-            host_config.memory = Some((ram * 1024. * 1024. * 1024.) as i64);
+            host_config.memory_reservation = Some((ram * 1024. * 1024. * 1024.) as i64);
         }
 
-        if let Some(cpu) = resources.cpu() {
-            host_config.cpu_count = Some(cpu as i64);
```

Comment:
This was a Windows-only field; if we want to limit the CPU allocation for the container, we should be using `nano_cpus`.

Reply:
Okay, today I learned.